### PR TITLE
Fix EOF bug

### PR DESCRIPTION
**Pull request by habnabit**
Also a few minor changes.

This was a bug exposed by this source from [gyp](https://github.com/nodejs/node-gyp):

```python
    generated_manifest_outer = \
"<?xml version='1.0' encoding='UTF-8' standalone='yes'?>" \
"<assembly xmlns='urn:schemas-microsoft-com:asm.v1' manifestVersion='1.0'>%s" \
"</assembly>"
```

### CODE REVIEW

File: ebb_lint/flake8.py [hunk 1]
```diff
@@ -273,6 +273,7 @@ def _check_tree(self, tree):
                 if not pattern.match(node, results):
                     continue
                 for k in extra.get('comments_for', ()):
+                    # XXX: this doesn't use `k` for finding the node
```
Comment (brighid) [inline ebb_lint/flake8.py]: What does it mean to use the `k` in `for k in extra.get('comments_for', ()):` to find a node?

Comment (habnabit) [inline ebb_lint/flake8.py]: `k` is supposed to name a specific node, but it isn't used for selecting a node when adding things to the results. I'll change the comment to make that clearer.

File: ebb_lint/test/test_flake8.py [hunk 2]
```diff
@@ -400,6 +400,30 @@ class C(object):
...
 
     ''',
+
+    # Oh boy these following two cases need explanation. The prefix of the
+    # 'spam' node in both of these cases ends with a literal \ followed by a
+    # newline, which cannot be tokenized by generate_tokens because it expects
+    # something to follow that sequence. This is one hell of a special case,
+    # but it made flake8 fail hard on valid syntax. The first case is a prefix
+    # containing no comment, and the second case also includes a comment which
+    # requires scanning.
```
Comment (brighid) [inline ebb_lint/test/test_flake8.py]: That's the line-continuation syntax, yeah? What's the particular part that's making it choke, the part where it's right after an assignment operator?

Comment (habnabit) [inline ebb_lint/test/test_flake8.py]: No. The assignment doesn't matter; it's just that the string literal node's prefix ends with `'\\\n'`, and a string passed to `generate_tokens` can't end with that. A node's prefix is every non-token character before the node, such as whitespace, comments, and line continuation characters.

Comment (brighid) [inline ebb_lint/test/test_flake8.py]: Oh, okay, got it. Jeeze, that _is_ pretty narrow.

Comment (jayvdb) [inline ebb_lint/test/test_flake8.py]: @habnabit , does this occur on all versions of Python?  If it is specific versions, this blurb should also note that.  If it is still occurring in Python 3.5, IMO it sounds like a bug that needs to be raised upstream. 

Not really relevant, but I note that `generate_tokens` is no longer in the [documented interface](https://docs.python.org/3/library/tokenize.html).  Very strangely, the [v2 docs](https://docs.python.org/2/library/tokenize.html) say `tokenize.generate_tokens` is the primary interface, and note "An older entry point is retained for backward compatibility: tokenize.tokenize".  It looks like the v2 docs are wrong, and `tokenize.tokenize` is now the preferred interface.

Comment (habnabit) [inline ebb_lint/test/test_flake8.py]: @jayvdb This is actually using `lib2to3.pgen2.tokenize.generate_tokens`. I don't think it's a bug in `generate_tokens`, though; it is, in fact, an incomplete document. The only reason I'm using `generate_tokens` at all is to find comment tokens.

**Pull request merged**